src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -5572,6 +5572,387 @@ theorem msb_replicate {n w : Nat} {x : BitVec w} :
   simp only [BitVec.msb, getMsbD_replicate, Nat.zero_mod]
   cases n <;> cases w <;> simp
 
+/-! ### Leading zeroes -/
+
+@[simp]
+theorem clzAux_zero {x : BitVec w} : clzAux x 0 = if x.getLsbD 0 then 0 else 1 := by simp [clzAux]
+
+theorem clzAuz_eq_zero_iff {x : BitVec w} {n : Nat}:
+    clzAux x n = 0 ↔ x.getLsbD n = true := by
+  unfold clzAux
+  cases n <;> simp
+
+theorem clzAux_le {x : BitVec w} {n : Nat} :
```

Comment:
Given that this is not an equality, should we make `x` and `n` non-optional arguments?

Reply:
closing bc outdated